check_test.go

```diff
@@ -98,11 +98,38 @@ var _ = Describe("Check", func() {
 				mirror.Close()
 			})
 
+			Context("when the repository contains a registry host name prefixed image", func() {
+				BeforeEach(func() {
+					mirror.AppendHandlers(
+						ghttp.CombineHandlers(
+							ghttp.VerifyRequest("GET", "/v2/"),
+							ghttp.RespondWith(http.StatusOK, `welcome to zombocom`),
+						),
+						ghttp.CombineHandlers(
+							ghttp.VerifyRequest("HEAD", "/v2/some/fake-image/manifests/latest"),
+							ghttp.RespondWith(http.StatusOK, `{"fake":"manifest"}`, http.Header{
+								"Docker-Content-Digest": {LATEST_FAKE_DIGEST},
+							}),
+						),
+					)
+
+					req.Source.Repository = mirror.Addr() + "/some/fake-image"
+					req.Source.RegistryMirror = &resource.RegistryMirror{
+						Host: "thisregistrymirrordoesnotexist.nothing",
+					}
+				})
+
+				It("it checks and returns the current digest using the registry declared in the repository and not using the mirror", func() {
+					Expect(res).To(Equal([]resource.Version{
+						{Digest: LATEST_FAKE_DIGEST},
+					}))
+				})
+			})
+
 			Context("which has the image", func() {
 				Context("in an explicit namespace", func() {
 					BeforeEach(func() {
-						// use the mock mirror as the "origin", use Docker Hub as a "mirror"
-						req.Source.Repository = mirror.Addr() + "/" + req.Source.Repository
+						// use Docker Hub as a "mirror"
```

Comment:
I think we can probably just delete some of these `in an explicit namespace` contexts since the above test kind of makes them redundant, but I can just do that in a followup commit since I've got work in progress in this part of the code anyway. 👍

Reply:
Thanks @vito - I appreciate the patience on all this. And I have a naive question: is `registry-image-resource:latest` continuously published to Docker Hub on merge to `master`? Or do I need to wait for the next Concourse release before this is published to Docker Hub? Or some other release event?

Reply:
Likewise!

Images are built from `master` and pushed to `concourse/registry-image-resource:dev-ubuntu` and `concourse/registry-image-resource:dev-alpine` by [this pipeline](https://ci.concourse-ci.org/teams/main/pipelines/registry-image).

I could also publish a new version but I think I'll wait until I've made this same fix here: https://github.com/concourse/docker-image-resource/pull/308 - so I can ship it all together. In any case there'll be a new version shipped and included in v6.7.0.